SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -449,19 +449,12 @@ void ActorSystem::HandleActorAuthority(const Worker_EntityId EntityId, const Wor
 						Actor->SetReplicates(true);
 					}
 
-					if (Actor->IsA<APlayerController>())
+					if (Channel->IsAutonomousProxyOnAuthority())
 					{
 						Actor->RemoteRole = ROLE_AutonomousProxy;
 					}
-					else if (APawn* Pawn = Cast<APawn>(Actor))
-					{
-						// The following check will return false on non-authoritative servers if the PlayerState hasn't been received yet.
-						if (Pawn->IsPlayerControlled())
```

Comment:
If this is fixed now, doesn't that mean we can remove the PlayerState check also?

Reply:
That's a good point, I'm not really sure what PlayerState does -- do you know enough to know how I can test that this case would still work properly?